### ISSUE TITLE
Support UX303LN

### ISSUE THREAD

**Issue by remychvn** [hieplpvip/ASUS-ZENBOOK-UX410-HACKINTOSH#3]
I have a Zenbook ux303LN and I try to make it a perfect hackintosh.
Here are the specs:

Intel Core i5 Broadwell 5200U
Intel Graphics HD5500
**discrete graphics**: Nvidia 940M (not working, must be disabled)
Audio Codec: Conexant CX20751/2
3x USB3.0 (Intel)
Automatic Light Sensor (ALS)
1x Full-size HDMI 2.0
1x Mini DisplayPort
Backlit Keyboard
3.5mm jack
FocalTech Trackpad
Intel 9 Chipset (Wildcat Point-LP SATA Controller etc)
Extra mSATA port for a half-size mSATA SSD inside
Intel Wireless-AC 7265 (M.2 NGFF card)

Here are the dsdt extracted with clover:
https://we.tl/X9uTWVgzUY

An old guide for an almost similar model. (Without dedicated graphics card)
https://www.tonymacx86.com/threads/guide-asus-zenbook-ux303la-broadwell-edition.172279/

Do you think you can add it to the list of supported computers?

Thanks

**Comment by hieplpvip** [comment]
It would take time since your laptop is an old one and is much different from mine.

For now, I recommend following that guide to get working macOS on your laptop.
Then I'll test my build and check what works and what not for you.

**Comment by remychvn** [comment]
I've tried and I'm getting a pretty good result but there are still important concerns for using it in daily use.

The PC will restart in a loop if the HDMI is plugged at boot. No problem if you plug it in once the PC has started.

I can't get the external microphones to work.

I'm sure someone more experienced could fix those bugs.

**Comment by hieplpvip** [comment]
Send me output from this [tool](https://www.tonymacx86.com/threads/tool-generate-proper-problem-reporting-files.235953/)

**Comment by hieplpvip** [comment]
Try [Codec Commander](https://bitbucket.org/RehabMan/os-x-eapd-codec-commander/downloads/) and [SSDT-CX20752](https://bitbucket.org/RehabMan/os-x-eapd-codec-commander/src/6b9514532920f2191c3863c67660ba42f934f350/SSDT-CX20752.dsl?at=master&fileviewer=file-view-default)

**Comment by remychvn** [comment]
Hello,

Thanks for your help !

This is the result of the extraction:
[debug_1461.zip](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/files/2268129/debug_1461.zip)

Panic log : Kernel_2018-08-07-232001 is the caused by connecting the HDMI before mac os starts.

Your patch solves my mic problems at least partially. The sound level remains very low even when the sound is set to maximum. (The setting does not seem to have any impact)
(Codec commander + applealc + lilu + your patch)

If I can solve this problem, it will start to be usable on a daily basis. There are still problems with sleep and the HDMI problem (not very important)

**EDIT** : Finally the sound recording via the connected microphone does not work. I think I saw it work. But since the external mic/line mic switch is not done correctly each time. Maybe I got screwed. What I must have seen working was the internal microphone, which explains the low sound.

**Comment by hieplpvip** [comment]
Try WhateverGreen.
Install CodecCommander to /Library/Extensions and remove it from Clover

**Comment by hieplpvip** [comment]
[CLOVER.zip](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/files/2271626/CLOVER.zip)
Use that to boot. Remember to prepare a bootable USB in case of kernel panic

**Comment by remychvn** [comment]
Thank you! 

Unfortunately it starts correctly but it doesn't seem to fix any of the bugs.
I also moved codecCommander in S/L/E and rebuilt the cache. But but it doesn't seem to have any impact. The only thing that appeared was audio via HDMI. (I don't know if it's due to "codec commander " or the new kexts)
[debug_20313.zip](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/files/2272317/debug_20313.zip)

**Comment by hieplpvip** [comment]
HDMI audio is the result of using WhateverGreen.
To fix external mic, I need codec dump from Linux. Follow this guide for how to do it: http://osxarena.com/2015/09/getting-codec-information-for-applehda-patching/

Did you check screen backlight? I saw You hadn't installed it in your build, so I have added new kext AppleBacklightFixup, hope it would give you backlight control.

**Comment by hieplpvip** [comment]
Btw, you should use a supported Wifi card like DW1560 instead of USB adapter

**Comment by hieplpvip** [comment]
[CLOVER.zip](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/files/2273427/CLOVER.zip)
Try this

**Comment by remychvn** [comment]
I didn't check it out. But I think it was already working. (except via the adjustment keys but with the slider yes). I'll check all this out.
Concerning the audio codec dump: I recently posted it on the AppleHDA patcher thread.
There he is:
[codec_dump.txt](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/files/2273892/codec_dump.txt)

Yes for the usb wifi card I saw that this could cause standby problems. But I test the standby by disconnecting before the wifi card each time.

I don't want to replace the wifi card if I can't make it a hackintosh for everyday use.

I'll try that in a few hours.

Thanks again for your time!

**Comment by hieplpvip** [comment]
Replacing wifi card is the first step for everyday use.

**Comment by hieplpvip** [comment]
[CLOVER.zip](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/files/2274690/CLOVER.zip)
Skip that, try this. Hope it fix audio

**Comment by remychvn** [comment]
Backlight setting with fn keys is working with your Clover folder. Thank you !
Unfortunately, the audio output / input is no longer detected at all.
Do you want a new debug extract ?

**Comment by hieplpvip** [comment]
Yes

**Comment by hieplpvip** [comment]
I have looked at your codec dump. You likely need a new AppleHDA patch.

**Comment by hieplpvip** [comment]
[CLOVER.zip](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/files/2277276/CLOVER.zip)
Try this

**Comment by remychvn** [comment]
Yes but the developer of the patch tool seems to say that there is no solution... http://olarila.com/forum/viewtopic.php?t=6528&start=20

I'll test with your Clover anyway.

**Comment by remychvn** [comment]
No input / output sound with latest clover folder.
[debug_32218.zip](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/files/2278981/debug_32218.zip)

**Comment by hieplpvip** [comment]
We should switch to gitter: http://gitter.im/hieplpvip

**Comment by hieplpvip** [comment]
[CLOVER-5.zip](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/files/2279784/CLOVER-5.zip)
Try this. Note that it may break something, I want to fix audio first, then we'll fix the rest.

Whenever you have time, DM me on gitter. I'm GMT+7.

**Comment by hieplpvip** [comment]
[CLOVER 2.zip](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/files/2279946/CLOVER.2.zip)
If the above CLOVER works, try this one too

**Comment by remychvn** [comment]
External microphone  is working with CLOVER-5.zip ! 
The weird thing is that it only works if the microphone is plugged in before the boot... I didn't test this way with the other clover folders. I was very lucky to notice that.

The microphone is not linked to the right sound input it seems to me.
![screen shot 2018-08-11 at 20 36 58](https://user-images.githubusercontent.com/5863606/43994985-916a40ea-9da6-11e8-94ce-4f58cfbfc7aa.png)

I'm gonna test the clover-2.zip now.

**Comment by remychvn** [comment]
With clover-2.zip external microphone doesnt work.
Finally it does not come from connecting the microphone before starting. It's completely random. Sometimes it works, sometimes it doesn't. It varies between starts without changing anything.

So I did 2 extractions:

The external microphone works:
[debug_25125.zip](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/files/2280482/debug_25125.zip)

The external microphone does not work:
[debug_7512.zip](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/files/2280484/debug_7512.zip)

Maybe it can help...

**Comment by bramand** [comment]
Hello,
I have almost similar laptop, my laptop is Zenbook UX303LB with specification:

- Intel Core i7 Broadwell 5500U
- Intel Graphics HD5500 (with dedicated graphics card, Nvidia GeForce 940M)
- Audio Codec: Conexant CX20751/2
- 1080p matte screen
- 3x USB3.0 (Intel)
- 1x SD Card reader
- WebCam
- Automatic Light Sensor (ALS)
- 1x Full-size HDMI 2.0
- 1x Mini DisplayPort
- Backlit Keyboard
- 3.5mm jack
- FocalTech Trackpad
- Intel 9 Chipset (Wildcat Point-LP SATA Controller etc)
- Extra mSATA port for a half-size mSATA SSD inside
- Comes with 8GB of RAM.
- Comes with two accessories: A miniDP to VGA converter, and a USB Ethernet dongle.
- Intel Wireless-AC 7265 (M.2 NGFF card), and I already changed to Broadcom BCM94352Z (DW-1560)

I already follow the guide from https://www.tonymacx86.com/threads/guide-asus-zenbook-ux303la-broadwell-edition.172279/, and i also patch dsdt and ssdt by myself. All of the above are working for we, i can use it in daily use, only sleep that not working, i try to discrete my nvidia graphics but always fail. then i saw your post. my question can i use your config.plist and your ssdt files from ux303ln to my laptop ux303lb with difference in processor type and wireless adapter? thank you

**Comment by hieplpvip** [comment]
@bramand Yes, try my files and report what works. Remember to create a recovery USB in case things go wrong

**Comment by bramand** [comment]
Hello,
Thank you for you answer.
I have already try your files names Clover 2.zip. i can boot to macOS Mojave, but some of things not work. that is battery indicator, backlit keyboard, discrete graphics card. because of that i assume that it's different between ux303la, ux303lb, ux303ln. Would you mind to check my EFI folder(includes pathced and original DSDT/SSDT and config.plist) that i have already pathced (only sleep that did not work). it based from the guide from https://www.tonymacx86.com/threads/guide-asus-zenbook-ux303la-broadwell-edition.172279/. then i want to know, better use hotpatching or use general patching ssdt/dsdt? because i only know how to patch use general pathing.

[EFI Zenbook UX303LB.zip](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/files/2507348/EFI.Zenbook.UX303LB.zip)

**Comment by hieplpvip** [comment]
@bramand Can you follow the [instruction](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/wiki/Basic-Instruction) to reinstall macOS?

**Comment by bramand** [comment]
@hieplpvip yes, before i try your file names Clover 2.zip. I have already follow your [instruction](https://github.com/hieplpvip/ASUS-ZENBOOK-HACKINTOSH/wiki/Basic-Instruction) and try your config.plist, kext, and patched files for ux303l. After i try that file i also get some of things that not work.

**Comment by hieplpvip** [comment]
@bramand Can you upload your problem reporting files when using my build (following the instruction)?

**Comment by WONGWAI1017** [comment]
> I have a Zenbook ux303LN and I try to make it a perfect hackingtosh.
> Here are the specs:
> 
> Intel Core i5 Broadwell 5200U
> Intel Graphics HD5500
> **discrete graphics**: Nvidia 940M (not working, must be disabled)
> Audio Codec: Conexant CX20751/2
> 3x USB3.0 (Intel)
> Automatic Light Sensor (ALS)
> 1x Full-size HDMI 2.0
> 1x Mini DisplayPort
> Backlit Keyboard
> 3.5mm jack
> FocalTech Trackpad
> Intel 9 Chipset (Wildcat Point-LP SATA Controller etc)
> Extra mSATA port for a half-size mSATA SSD inside
> Intel Wireless-AC 7265 (M.2 NGFF card)
> 
> Here are the dsdt extracted with clover:
> https://we.tl/X9uTWVgzUY
> 
> An old guide for an almost similar model. (Without dedicated graphics card)
> https://www.tonymacx86.com/threads/guide-asus-zenbook-ux303la-broadwell-edition.172279/
> 
> Do you think you can add it to the list of supported computers?
> 
> Thanks

Same model here. May I get your EFI files?
